### PR TITLE
Fix invalid local uploads path

### PR DESCRIPTION
**Pull request by akcyp**
File upload provider does not consider the strapi configuration. This patch should fix the problem.

#### Description of what you did:
```js
// app.js
const path = require('path');
const strapiDir = path.join(__dirname, './backend');

const Strapi = require(path.join(strapiDir, `/node_modules/strapi`));
const strapi = Strapi({
  dir: strapiDir
});
strapi.start();
```
This configuration generates the following error when uploading the image:
```
ENOENT: no such file or directory, open 'C:\Users\XYZ\Desktop\backend\public\uploads\screenshot_862571e640.jpeg'
```
The correct directory should be:
```
C:\Users\XYZ\Desktop\app\backend\public\uploads\screenshot_862571e640.jpeg
```

### CODE REVIEW

**Comment by derrickmehaffy** [comment]
This will need to be checked against Mac and Linux

File: packages/strapi-provider-upload-local/lib/index.js [hunk 1]
```diff
@@ -23,6 +23,10 @@ module.exports = {
         });
       }
     };
+    const getStaticPath = () => path.join(
```
Comment (alexandrebodin) [inline packages/strapi-provider-upload-local/lib/index.js:26]: this doesn't need to be a function. you can use 
```js
const uploadDir = path.join(strapi.dir, strapi.config.middleware.settings.public.path || strapi.config.paths.static)
```

**Comment by akcyp** [comment]
You mean `packages/strapi-plugin-upload/test/__tests__/bootstrap.test.js` ? 😄 
I made changes but when running `jest ./packages/strapi-plugin-upload/test/__tests__/bootstrap.test.js` tests failed 😢 
The error `ReferenceError: register is not defined` occurs at line
```js
services: { permission: { actionProvider: { register } } },
```

**Comment by alexandrebodin** [comment]
@akcyp You need to update your branch (merge master) and then follow the previous comment ;)

**Comment by akcyp** [comment]
Thanks and forgive me for making corrections now, but I was on vacation. I hope everything will be ok now! 💯 😄

**Comment by moritz-orth** [comment]
Hi, with 3.0.* we had configured `paths.static` to use a absolute path (e.g. `/mnt/storage/uploads`) instead of the default `./uploads` location. With this change the strapi.dir always prepends the `pwd()` location, thus screwing with our config.

As a workaround, cloned the local provider (making a custom provider) and removed the strapi.dir prefix. But I'm not sure if this is the best approach. 

**Is there a better way to configure an absolute path for the upload folder?**

**Comment by alexandrebodin** [comment]
@moritz-halke a fix is ready for the next release: https://github.com/strapi/strapi/pull/7280

**Pull request merged**